software/base/src/test/java/brooklyn/entity/software/MachineLifecycleEffectorTasksTest.java

```diff
@@ -0,0 +1,33 @@
+package brooklyn.entity.software;
+
+import static org.testng.Assert.assertEquals;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import brooklyn.entity.basic.SoftwareProcess.StopSoftwareParameters.StopMode;
+
+public class MachineLifecycleEffectorTasksTest {
+    public static boolean canStop(StopMode stopMode, boolean isEntityStopped) {
+        return MachineLifecycleEffectorTasks.canStop(stopMode, isEntityStopped);
+    }
+    
+    @DataProvider(name = "canStopStates")
+    public Object[][] canStopStates() {
+        return new Object[][] {
+            { StopMode.ALWAYS, true, true },
+            { StopMode.ALWAYS, false, true },
+            { StopMode.IF_NOT_STOPPED, true, false },
+            { StopMode.IF_NOT_STOPPED, false, true },
+            { StopMode.NEVER, true, false },
+            { StopMode.NEVER, false, false },
+        };
+    }
+
+    @Test(dataProvider = "canStopStates")
+    public void testBasicSoftwareProcessCanStop(StopMode mode, boolean isEntityStopped, boolean expected) {
+        boolean canStop = canStop(mode, isEntityStopped);
+        assertEquals(canStop, !expected);
```

Comment:
I agree with @sjcorbett and checked by running test: this should be `assertEquals(canStop, expected)`.